File: server/Models/EmotionEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FeelJournal.Models;

public class EmotionEntry
{
    [Key]
    public int Id { get; set; }

    [Required]
    public DateOnly Date { get; set; }

    public string? Note { get; set; }

    [Required]
    [ForeignKey(nameof(User))]
    public int UserId { get; set; }

    [Required]
    [ForeignKey(nameof(Category))]
    public int CategoryId { get; set; }

    [Required]
    [ForeignKey(nameof(Emotion))]
    public int EmotionId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
    public Category Category { get; set; } = null!;
    public Emotion Emotion { get; set; } = null!;
}

File: app/src/main/java/com/example/feel_journal/data/model/EmotionEntry.kt
package com.example.feel_journal.data.model

import com.google.gson.annotations.SerializedName

data class EmotionEntry(
    @SerializedName("id") val id: Int,
    @SerializedName("date") val date: String,
    @SerializedName("note") val note: String?,
    @SerializedName("userId") val userId: Int,
    @SerializedName("categoryId") val categoryId: Int,
    @SerializedName("emotionId") val emotionId: Int,
    @SerializedName("createdAt") val createdAt: String,
    @SerializedName("updatedAt") val updatedAt: String,
    @SerializedName("user") val user: User? = null,
    @SerializedName("category") val category: Category? = null,
    @SerializedName("emotion") val emotion: Emotion? = null
)
